jupyter/muehle-game.ipynb

```diff
@@ -222,6 +222,42 @@
     "    else:\n",
     "        return nextStatesPhaseThree(s, p)"
    ]
+  },
+  {
+   "source": [
+    "## Game Progess"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def finished(s):\n",
+    "    return not (hasEnoughtStones(s, 'w') and \\\n",
+    "                hasEnoughtStones(s, 'b') and \\\n",
+    "                len(nextStates(s, 'w')) > 0 and \\\n",
+    "                len(nextStates(s, 'b')) > 0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def utility(s, p):\n",
+    "    utility  = -1 if not hasEnoughtStones(s, p) else 0\n",
+    "    utility +=  1 if not hasEnoughtStones(s, opponent(p)) else 0\n",
+    "    if utility != 0:\n",
+    "        return utility\n",
+    "    utility  = -1 if len(nextStates(s, p)) <= 0 else 0\n",
```

Comment:
also ich hab mir die funktion `utility` mal angeschaut. Funktioniert alles wie es soll. 🚀 

Allerdings verstehe ich die Vergleiche der Länge der nextStates auf kleiner gleich 0 nicht. Also nach 10 sekunden Google habe ich keine Möglichkeit gefunden, wann eine Liste in Python eine negative Länge haben kann (man weiß ja nie bei Python xD). Deswegen verwirrt mich das eher, als einfach auf `== 0` zu überprüfen. ^^

Reply:
Haha, ja das geht nicht :D Ich habe immer schlechte Erfahrung gemacht, wenn man bei Zahlen auf Gleichheit prüft, zwar eigentlich nur wenn es floats sind, aber deswegen hat sich das bei mir so eingebürgert. Wenn du willst kann ich das ändern?

Reply:
Jau ich weiß was du meinst, aber bei einer Listenlänge ist es halt irgendwie komisch imo. Aber wegen mir kannst du es auch lassen ^^

Reply:
Haha mir wäre gerade noch eine richtig unübersichtliche Lösung eingefallen die das Problem umgeht:
```python
utility  = -1 * max(len(nextStates(s, p), 1)
utility +=  1 * max(len(nextStates(s, opponent(p)), 1)
```
Aber das lassen wir mal lieber :D